### PR TITLE
feat: Add a config to disable automatic Wynncraft resource pack loading for the Wynncraft button

### PR DESCRIPTION
**Pull request by sofiedotcafe**
#1630

Added an option to disable automatic Wynncraft resource pack loading
As an example this helps e.g. Olinus10's modpack so you dont need to go though the server list menu!

- jh-devv

### CODE REVIEW

File: common/src/main/resources/assets/wynntils/lang/en_us.json [hunk 1]
```diff
@@ -890,6 +890,8 @@
   "feature.wynntils.wybelSound.name": "Wybel Sounds",
   "feature.wynntils.wynncraftButton.autoConnect.description": "Should the mod auto-connect to Wynncraft on startup?",
   "feature.wynntils.wynncraftButton.autoConnect.name": "Auto-Connect on Startup",
+  "feature.wynntils.wynncraftButton.autoLoadResourcePack.description": "Should the Wynncraft texture pack be auto loaded?",
+  "feature.wynntils.wynncraftButton.autoLoadResourcePack.name": "Auto load resource pack",
```
Comment (kristofbolyai) [inline common/src/main/resources/assets/wynntils/lang/en_us.json]:
```suggestion
  "feature.wynntils.wynncraftButton.autoLoadResourcePack.name": "Load Resource Pack",
```

File: common/src/main/resources/assets/wynntils/lang/en_us.json [hunk 2]
```diff
@@ -890,6 +890,8 @@
   "feature.wynntils.wybelSound.name": "Wybel Sounds",
   "feature.wynntils.wynncraftButton.autoConnect.description": "Should the mod auto-connect to Wynncraft on startup?",
   "feature.wynntils.wynncraftButton.autoConnect.name": "Auto-Connect on Startup",
+  "feature.wynntils.wynncraftButton.autoLoadResourcePack.description": "Should the Wynncraft texture pack be auto loaded?",
```
Comment (kristofbolyai) [inline common/src/main/resources/assets/wynntils/lang/en_us.json]:
```suggestion
  "feature.wynntils.wynncraftButton.autoLoadResourcePack.description": "Should the Wynncraft resource pack be loaded?",
```

File: common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java [hunk 3]
```diff
@@ -46,6 +46,9 @@ public class WynncraftButtonFeature extends Feature {
     @RegisterConfig
     public final Config<Boolean> autoConnect = new Config<>(false);
 
+    @RegisterConfig
+    public final Config<Boolean> autoLoadResourcePack = new Config<>(true);
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java]: Can you name this `loadResourcePack` as it is more fitting? You basically either load or not load the pack, there is no "auto" in there.

File: common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java [hunk 4]
```diff
@@ -80,7 +83,12 @@ private void addWynncraftButton(TitleScreen titleScreen) {
     private ServerData getWynncraftServer() {
         ServerData wynncraftServer =
                 new ServerData("Wynncraft", connectToLobby.get() ? LOBBY_SERVER : GAME_SERVER, false);
-        wynncraftServer.setResourcePackStatus(ServerData.ServerPackStatus.ENABLED);
+
+        if (autoLoadResourcePack.get()) {
+            wynncraftServer.setResourcePackStatus(ServerData.ServerPackStatus.ENABLED);
+        } else {
+            wynncraftServer.setResourcePackStatus(ServerData.ServerPackStatus.DISABLED);
+        }
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java]: Could you refactor this to use [Ternary operations](https://www.geeksforgeeks.org/java-ternary-operator-with-examples/)?

**Review by kristofbolyai — COMMENTED** [review]
Looks good, this could work, but I want you to do some renaming and refactoring to make both the code and the user representation clearer.

File: common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java (hunk 3 above)
Comment (sofiedotcafe) [reply common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java]: Okay, i can do that

File: common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java (hunk 4 above)
Comment (sofiedotcafe) [reply common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java]: Okay

**Comment by sofiedotcafe** [comment]
Is it good now? @kristofbolyai?

**Comment by sofiedotcafe** [comment]
Did some final typo fixing

File: common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java [hunk 5]
```diff
@@ -80,7 +83,9 @@ private void addWynncraftButton(TitleScreen titleScreen) {
     private ServerData getWynncraftServer() {
         ServerData wynncraftServer =
                 new ServerData("Wynncraft", connectToLobby.get() ? LOBBY_SERVER : GAME_SERVER, false);
-        wynncraftServer.setResourcePackStatus(ServerData.ServerPackStatus.ENABLED);
+
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java]:
```suggestion
```

**Comment by sofiedotcafe** [comment]
Now good?

**Review by kristofbolyai — APPROVED** [review]
Looks good now! Thanks.

**Pull request merged**